### DISCUSSION TITLE
How to integrate into a custom ROMs settings

### DISCUSSION THREAD

**Discussion by LeOS-GSI** [seedvault-app/seedvault#395] [Ideas]
How can I access Seedvault for initial start /or manual backup start with adb command or secret code like restoring ?
In my LeOS-S (Android12.1) I don't have a Option in Settings/System/Backup to start Seedvault.

**Comment by ChaosNicro** [comment]
Potentially include a boolean or other config item to grant apps the permission to open the "SettingsActivity" that is exported by Seedvault, or let them ask the user to grant it, like some signature-spoof implementations do it. From what I understand the activity has the same effect as the menu-item in settings.
Adding a second number-code to "SecretCodeReceiver.kt" is of course also an option.

**Comment by chirayudesai** [comment]
Secret code already exists, see https://github.com/seedvault-app/seedvault/wiki/FAQ#how-to-restore-after-passing-through-setup-wizard

**Reply by ChaosNicro** [reply]
The secret code only works for the restore activity, we are talking about opening the menu or triggering/setting up a backup without having access to the seedvault entry in settings.

**Reply by LeOS-GSI** [reply]
> Secret code already exists, see https://github.com/seedvault-app/seedvault/wiki/FAQ#how-to-restore-after-passing-through-setup-wizard

But only for restoring, not for backup.

**Comment by chirayudesai** [comment]
@LeOS-GSI you should set seedvault as backup provider.

If you're shipping AOSP without GMS / only shipping Seedvault, see https://github.com/seedvault-app/seedvault/wiki/Building-with-AOSP (step 8) - it can be a RRO as well.

If you're shipping with GMS / want a picker, you can pick the patches from LineageOS

**Reply by LeOS-GSI** [reply]
https://github.com/seedvault-app/seedvault/wiki/Building-with-AOSP -- This was only working til A12. Since A12.1 it isn't working anymore. No way to set it up as backup provider. Restore with hidden number and adb are working fine. But setup a new backup isn't working. There should be also a adb command or hidden number as I have written in my initial post. I will try it with creating a hidden number for backup, but I haven't had enough time in the moment

**Reply by chirayudesai** [reply]
> Since A12.1 it isn't working anymore. No way to set it up as backup provider.

Works fine here, what exactly happens?

> There should be also a adb command

`adb shell bmgr` a command under here should let you set seedvault as default transport, but that should not be needed with the patch.

> I will try it with creating a hidden number for backup

Not something we're likely to merge, as that's the wrong way to fix this problem.

**Reply by LeOS-GSI** [reply]
When I setup the point 8 in the A12.1 Android sources (I'm building GSIs) the complete GUI will break, Several Seeting menu options are gone, the recent button is gone and and and. The overlay is breaking all. In A10, A11, A12 GSIs I haven't had any issue

EDIT: have just tried 'adb shell bmgr'. That's working fine and I will use it as workaround

**Reply by LeOS-GSI** [reply]
By the way, these GUI glitshes are only happens when dirty flash an A12.1 GSI over an A12 GSI. With a clean flash it is working.

**Reply by LeOS-GSI** [reply]
I have just had a look in the AOSP sources. The right path to the file is 
frameworks/base/packages/SettingsProvider/res/values/default.xml 
and not
overlay/common/frameworks/base/packages/SettingsProvider/res/values/default.xml
the overlay path is for LineageOS sources

in line 53/54 the definition is done

and this https://gitlab.com/calyxos/platform_frameworks_base/commit/9738a0c752dd75b1c6146882e2ca53a2232e7148 is very old. in the meantime we are at currentVersion 205

**Reply by chirayudesai** [reply]
> The right path to the file is

Fixed!

>  in the meantime we are at currentVersion 205

We haven't used this patch in a while, that's why. You can simply bump the version, the rest of it should be the same and still work. It says so in the wiki

It's only required when going from a build without seedvault enabled (like yours) to one with seedvault enabled. For fresh installs it's not needed.
